app/settings/data-export/data-export.controller.js

```diff
@@ -54,6 +56,13 @@ function (
         }
     });
 
+    // Check if hxl-api-key is added
+    UserSettingsEndpoint.getFresh({id: $rootScope.currentUser.userId}).$promise.then((user) => {
+        if (user.count !== 0) {
+            $scope.hxlApiKey = true;
```

Comment:
Do we need to add a specific check for the presence of the hxl_api_key? I think this checks to see if there are user settings associated with the given user and if so then we don't need them to enter it again, otherwise we have to ask for it. Is that right or am I way off?

Reply:
I am just checking if there is a hxl_api_key associated by the user here. Am not using the key for anything else than i displaying a warning and disable a button if its not there.